fairmq/plugins/DDS/runDDSCommandUInew.cxx

```diff
@@ -0,0 +1,199 @@
+/********************************************************************************
+ *  Copyright (C) 2014-2019 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include <fairmq/sdk/commands/Commands.h>
+#include <fairmq/States.h>
+#include <fairmq/SDK.h>
+
+#include <boost/program_options.hpp>
+
+#include <termios.h> // raw mode console input
+#include <unistd.h>
+#include <condition_variable>
+#include <cstdlib>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <utility>
+
+using namespace std;
+using namespace fair::mq;
+using namespace fair::mq::sdk;
+using namespace fair::mq::sdk::cmd;
+namespace bpo = boost::program_options;
+
+struct TerminalConfig
+{
+    explicit TerminalConfig()
+    {
+        termios t;
+        tcgetattr(STDIN_FILENO, &t); // get the current terminal I/O structure
+        t.c_lflag &= ~ICANON; // disable canonical input
+        // t.c_lflag &= ~ECHO; // do not echo input chars
+        tcsetattr(STDIN_FILENO, TCSANOW, &t); // apply the new settings
+    }
+
+    ~TerminalConfig()
+    {
+        termios t;
+        tcgetattr(STDIN_FILENO, &t); // get the current terminal I/O structure
+        t.c_lflag |= ICANON; // re-enable canonical input
+        // t.c_lflag |= ECHO; // echo input chars
+        tcsetattr(STDIN_FILENO, TCSANOW, &t); // apply the new settings
+    }
+};
+
+void printControlsHelp()
+{
+    cout << "Use keys to control the devices:" << endl;
+    cout << "[c] check states, [o] dump config, [h] help, [r] run, [s] stop, [t] reset task, [d] reset device, [q] end, [j] init task, [i] init device, [k] complete init, [b] bind, [x] connect" << endl;
+    cout << "To quit press Ctrl+C" << endl;
+}
+
+void sendCommand(const string& commandIn, const string& topologyPath, Topology& topo)
+{
+    char c;
+    string command(commandIn);
+    TerminalConfig tconfig;
+
+    if (command == "") {
+        printControlsHelp();
+        cin >> c;
+        command = c;
+    }
+
+    while (true) {
+        if (command == "c") {
+            cout << "> checking state of the devices" << endl;
+            topo.GetCurrentState();
+            // TODO: extend me
+        } else if (command == "o") {
+            cout << "> dumping config of the devices" << endl;
+            auto const result = topo.GetProperties("^(session|id)$", topologyPath);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'result' is assigned a value that is never used.](https://app.codacy.com/manual/FairRootGroup/FairMQ/pullRequest?prid=5000642)